tests/flows/test_vesting.py

```diff
@@ -106,73 +103,535 @@ def test_main(
+):
+    """Tests failure of the pool creation for a lower vesting period."""
+
+    erc721_nft, erc20_token = deploy_erc721_erc20(
+        web3, config, consumer_wallet, consumer_wallet, cap=to_wei(1000)
+    )
+    initial_ocean_liq = to_wei(200)
+
     ocean_contract = ERC20Token(web3=web3, address=get_address_of_type(config, "Ocean"))
-    ocean_contract.approve(
-        get_address_of_type(config, "Router"), to_wei("0.02"), consumer_wallet
+    ocean_contract.approve(factory_router.address, initial_ocean_liq, consumer_wallet)
+
+    with pytest.raises(exceptions.ContractLogicError) as err:
+        erc20_token.deploy_pool(
+            rate=to_wei(1),
+            base_token_decimals=ocean_contract.decimals(),
+            vesting_amount=initial_ocean_liq // 100 * 9,
+            vesting_blocks=100,
+            base_token_amount=initial_ocean_liq,
+            lp_swap_fee_amount=to_wei("0.003"),
+            publish_market_swap_fee_amount=to_wei("0.001"),
+            ss_contract=get_address_of_type(config, "Staking"),
+            base_token_address=ocean_contract.address,
+            base_token_sender=consumer_wallet.address,
+            publisher_address=consumer_wallet.address,
+            publish_market_swap_fee_collector=get_address_of_type(
+                config, "OPFCommunityFeeCollector"
+            ),
+            pool_template_address=get_address_of_type(config, "poolTemplate"),
+            from_wallet=consumer_wallet,
+        )
+        assert (
+            err.value.args[0]
+            == "execution reverted: VM Exception while processing transaction: revert ERC20Template: Vesting period too low. See FactoryRouter.minVestingPeriodInBlocks"
+        )
+
+
+@pytest.mark.unit
+def test_vesting_main_flow(web3, config, consumer_wallet, factory_router, side_staking):
+    """Tests consumer deploys pool, checks fees and checks if vesting is correct after 100 blocks."""
+
+    erc721_nft, erc20_token = deploy_erc721_erc20(
+        web3, config, consumer_wallet, consumer_wallet, cap=to_wei(1000)
+    )
+
+    # Consumer deploys pool & checks fees
+    mint_fake_OCEAN(config)
+    vesting_amount = to_wei(18)
+    initial_ocean_liq = to_wei(200)
+    pmt_collector = erc20_token.get_payment_collector()
+
+    block_before_deploying_pool = web3.eth.block_number
+    bpool = _deploy_ocean_pool(
+        erc20_token=erc20_token,
+        factory_router=factory_router,
+        config=config,
+        web3=web3,
+        from_wallet=consumer_wallet,
+    )
+    block_deployed_pool = block_before_deploying_pool + 2
+    assert bpool.is_finalized()
+    assert bpool.opc_fee() == to_wei("0.001")
+    assert bpool.get_swap_fee() == to_wei("0.003")
+    assert bpool.community_fee(get_address_of_type(config, "Ocean")) == 0
+    assert bpool.community_fee(erc20_token.address) == 0
+    assert bpool.publish_market_fee(get_address_of_type(config, "Ocean")) == 0
+    assert bpool.publish_market_fee(erc20_token.address) == 0
```

Comment:
A function should always do one thing and do it well. This is the case for test functions too.

This function is about testing the main flow of vesting. Alright. 

What I've highlighted is a bunch of tests for fees. This is inappropriate for this function's target behavior. And inappropriate for this module. Please move those tests into the appropriate module.

Reply:
I have removed those line of code because there were already in `test_swap_fees.py`.